theano/tensor/opt.py

```diff
@@ -3077,6 +3077,22 @@ def local_useless_tile(node):
             return
 
 
+##############
+# Split Opts #
+##############
+@register_canonicalize
+@register_specialize
+@gof.local_optimizer([T.Split])
+def local_useless_split(node):
+    """ Split{n_splits=1}(x, y) -> x
+
+    Remove Split with only 1 split.
+    """
+    if isinstance(node.op, T.Split):
+        if node.op.len_splits == 1:
+            return [node.inputs[0]]
```

Comment:
To not remove the error, you can add an assert op in the graph:

return [tensor.assert_op(node.inputs[0], node.inputs[0].shape[node.inputs[1]] == node.inputs[2])]

We try to not remove user error anymore by default. It make for hard to find bugs in user code.

There is an opt to remove all the assert if you do not want them. You can use the Thaeno flags: optimizer_excluding='local_remove_all_assert' that will be abreviated soon to optimizer_excluding='unsafe'